### PR TITLE
Loosen validation for custom user data properties

### PR DESCRIPTION
**Pull request by orangejenny**
https://manage.dimagi.com/default.asp?270666

https://github.com/dimagi/commcare-hq/pull/18728/files#r154690363 ...famous last words.

Should go into today's deploy so this user gets unblocked.

@dannyroberts what do you think? I updated the validation to only check for the case that genuinely breaks (property names that are all numbers). Left the message saying 'Properties must start with a letter and consist only of letters, numbers, underscores or hyphens.' even though that's not what's being validated, because I still worry that we'll introduce code at some point that breaks if these user-entered property names aren't valid xml identifiers.

code buddy @gcapalbo

### CODE REVIEW

File: corehq/apps/custom_data_fields/edit_model.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ class XmlSlugField(forms.SlugField):
     default_validators = [
         validate_slug,
         validate_reserved_words,
-        RegexValidator(r'^[a-zA-Z]', ''),
+        RegexValidator(r'\D', ''),
```
Comment (emord) [inline corehq/apps/custom_data_fields/edit_model.py]: This is changing from "the string must begin with a letter" to "there is a non-digit in this string" is that correct? may be useful to add a comment/tests describing what this intended to catch.

If my understanding is correct then I approve.

Comment (orangejenny) [reply corehq/apps/custom_data_fields/edit_model.py]: yep, that's it. added comment.

File: corehq/apps/custom_data_fields/edit_model.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ class XmlSlugField(forms.SlugField):
     default_validators = [
         validate_slug,
         validate_reserved_words,
-        RegexValidator(r'^[a-zA-Z]', ''),
+        RegexValidator(r'\D', ''),  # disallow property names that are just numbers, which breaks
```
Comment (esoergel) [inline corehq/apps/custom_data_fields/edit_model.py]: I'm confused by both the before and after here - does `RegexValidator` raise an error if the value doesn't match the string?  Ie, would the old version succeed as long as there is any alphabetic character?  Does the new one succeed as long as there is any non numeric character present?  What about leading digits and invalid characters?

We do have this function which checks for valid xml identifiers:
https://github.com/dimagi/commcare-hq/blob/619c956550cd1409678163ff22083338e891ccd2/corehq/apps/fixtures/utils.py#L21-L21
A pretty robust approach would be to wrap that:
```python
def validate_xml_identifier(slug):
    if is_identifier_invalid(slug):
        raise ValidationError("message")
```

Comment (esoergel) [reply corehq/apps/custom_data_fields/edit_model.py]: Oh, `validate_slug` might take care of invalid characters.  Awesome new line linking there, github!

Comment (esoergel) [reply corehq/apps/custom_data_fields/edit_model.py]: Curious what'll happen if I link a code block:
https://github.com/dimagi/commcare-hq/blob/619c956550cd1409678163ff22083338e891ccd2/corehq/apps/fixtures/utils.py#L21-L37

**Pull request merged**